mediatek: filogic: upstream LEDs control on 2.5G ports on EX5601

abajk opened the pull request:
This commit switches the control of the leds connected to the Maxlinear GPY211C PHY to an upstream solution. The behaviour of LED1 is the same as before. The behaviour of 2.5G-WAN LED has been changed. It is only active when a 2.5G link is detected, which matches the stock software. Additionally, the name of the WAN led has been changed to INTERNET.

Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>

abajk committed:
mediatek: filogic: upstream LEDs control on 2.5G ports on EX5601

This commit switches the control of the leds connected to the Maxlinear
GPY211C PHY to an upstream solution. The behaviour of LED1 is the same
as before. The behaviour of 2.5G-WAN LED has been changed. It is only
active when a 2.5G link is detected, which matches the stock software.
Additionally, the name of the WAN led has been changed to INTERNET.

Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>
Link: https://github.com/openwrt/openwrt/pull/17952
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

dangowrt reviewed (approved):
LGTM. I don't know this device so I don't know why there are two LEDs both indicating eth1 link status, but I suppose that the PHY LEDs are on the back of the device while the "INTERNET" LED is on the front, right?

abajk commented:
> LGTM. I don't know this device so I don't know why there are two LEDs both indicating eth1 link status, but I suppose that the PHY LEDs are on the back of the device while the "INTERNET" LED is on the front, right?

The “2.5-WAN” and “INERNET” LEDs are on the front of the router next to each other. I haven't used the stock software, so I don't know how the internet LED behaves. I decided not to change its current behavior in OpenWRT.
The LEDs on the back:
LAN1 - green - gpy113c
LAN2 - green - mt7531,
LAN3 - green - mt7531,
LAN4 - green - mt7531.
LEDs on the front panel:
POWER - green and red - SoC,
2.5G-WAN - green - gpy113c,
INERNET - green and red - SoC,
LAN - green - SoC,
2.4G - green and amber - SoC,
5G - green and amber - SoC,
PHONE - green and amber - SoC,
SFP - green - SoC.

abajk commented:
Only a few devices use this hack to control LEDs:
~~~
abajkowski@abajkowski:~/workspace/openwrt/target/linux/mediatek$ grep -rni "mxl,led-config" .
./dts/mt7986a-zyxel-ex5700-telenor.dts:161:		mxl,led-config = <0x3f0 0x330 0x0 0x0>;
./dts/mt7986a-zyxel-ex5700-telenor.dts:167:		mxl,led-config = <0x3f0 0x330 0x0 0x0>;
./dts/mt7986a-asus-tuf-ax6000.dts:150:		mxl,led-config = <0x03f0 0x0 0x0 0x0>;
./dts/mt7986a-asus-tuf-ax6000.dts:158:		mxl,led-config = <0x03f0 0x0 0x0 0x0>;
./dts/mt7986a-acer-predator-w6d.dts:176:		mxl,led-config = <0x0 0x0 0x370 0x380>;
./dts/mt7986a-netcore-n60.dts:112:		mxl,led-config = <0x0 0x0 0x0 0x3f0>;
./dts/mt7986a-smartrg-bonanza-peak.dtsi:224:		mxl,led-config = <0x30 0x40 0x80 0x0>;
./dts/mt7986a-smartrg-bonanza-peak.dtsi:233:		mxl,led-config = <0x30 0x40 0x80 0x0>;
./dts/mt7981b-zyxel-nwa50ax-pro.dts:89:		mxl,led-config = <0x3b0 0x0 0x0 0x3c0>;
./dts/mt7986a-acer-predator-w6.dts:190:		mxl,led-config = <0x0 0x0 0x370 0x380>;
~~~
 It would be nice to migrate them all to stop users from adding new devices using this hack.
 Soon I will have access to an ASUS AX6000 then I will try to migrate it.

pameruoso commented:
Just my 2 cents here. Don't confuse the 2.5g WAN phy with the internet led and the actual 2.5g link led.
The 2.5g blue RJ45 port (the wan) has no led on the back. The led associated to that GPY is the link led in the front panel.
The internet led instead is a dedicated GPIO.  Personally I would not change this.

abajk commented:
> That is why in my initial commit i configured the wan phy led in the front to stay up without blinking only if the negotiated link is 2.5gbit. The link led indeed on the front has the 2.5g form.

@pameruoso Do you suggest any changes with my PR?

By the way. Do you know how the Internet LED behaves on stock software?

pameruoso commented:
> > That is why in my initial commit i configured the wan phy led in the front to stay up without blinking only if the negotiated link is 2.5gbit. The link led indeed on the front has the 2.5g form.
> 
> @pameruoso Do you suggest any changes with my PR?
> 
> By the way. Do you know how the Internet LED behaves on stock software?

On the original firmware the internet led on the front is green when ppoe is up ( at least here in Italy), red when down (no blinking).

The 2.5glink led on the front (managed by the phy powers on only if the negotiated link is 2.5gbit) 

I agree of course on standardizing the dts led management/syntax from your code. I m only pointing out that the front 2.5 should be powered on only when 2.5 link is negotiated. There is a dedicated led for internet and that stays fixed green when the ppoe interface is up.

abajk commented:
> The 2.5glink led on the front (managed by the phy powers on only if the negotiated link is 2.5gbit)
> 

This is a simple change and requires changing the trigger from "link" to "link_2500".


> I agree of course on standardizing the dts led management/syntax from your code. I m only pointing out that the front 2.5 should be powered on only when 2.5 link is negotiated.

This description does not match what I see in the [GPL sources](https://github.com/pameruoso/zyxel-ex5601t0/blob/e3077d117a9bdaf3af60da4fb88f119b05157523/target/linux/mediatek/files-5.4/arch/arm64/boot/dts/mediatek/ex5601-t0.dts#L203). The default setting is 240=0x00F0 which corresponds to no blinking and constant light when the link has a speed of 10/100/1000/2500. The datasheet for the GPY211 is [publicly available](https://www.maxlinear.com/product/interface/ethernet/ethernet-transceivers-phy/gpy211). Did I misunderstand something?

abajk commented:
> Anyways this is the led and it doesn’t make sense for me to power that on if the link is 10/100/1000. ![IMG_0784](https://private-

I have addressed the comment about the 2.5G-WAN LED.

pameruoso commented:
All good for me

GeorgeSapkin commented:
Is this going to be backported to 24.10?

systemcrash commented:
@GeorgeSapkin Create a cherry-pick PR for this to 24.10.
